commit f92a065483828ba172c858e6cbc46f75a8b933d9
Author: Paul Holzinger <pholzing@redhat.com>
Date:   Mon Jul 10 14:54:12 2023 +0200

    bump netlink deps
    
    There are breaking changes here so fix them, sigh.
    
    Signed-off-by: Paul Holzinger <pholzing@redhat.com>

Index: netavark/src/network/bridge.rs
===================================================================
--- netavark.orig/src/network/bridge.rs
+++ netavark/src/network/bridge.rs
@@ -471,7 +471,7 @@ fn create_interfaces(
         Ok(bridge) => check_link_is_bridge(bridge, &data.bridge_interface_name)?,
         Err(err) => match err.unwrap() {
             NetavarkError::Netlink(e) => {
-                if -e.code != libc::ENODEV {
+                if -e.raw_code() != libc::ENODEV {
                     // if bridge does not exists we will create it below,
                     // for all other errors we want to return the error
                     return Err(err).wrap("get bridge interface");
@@ -551,7 +551,7 @@ fn create_veth_pair(
     host_veth.info_data = Some(InfoData::Veth(VethInfo::Peer(peer)));
 
     host.create_link(host_veth).map_err(|err| match err {
-        NetavarkError::Netlink(ref e) if -e.code == libc::EEXIST => NetavarkError::wrap(
+        NetavarkError::Netlink(ref e) if -e.raw_code() == libc::EEXIST => NetavarkError::wrap(
             format!(
                 "create veth pair: interface {} already exists on container namespace",
                 data.container_interface_name
Index: netavark/src/network/netlink.rs
===================================================================
--- netavark.orig/src/network/netlink.rs
+++ netavark/src/network/netlink.rs
@@ -217,7 +217,7 @@ impl Socket {
             Ok(result) => result,
             Err(err) => match err {
                 // kernel returns EACCES when we try to add an ipv6 but ipv6 is disabled in the kernel
-                NetavarkError::Netlink(ref e) if -e.code == libc::EACCES => match addr {
+                NetavarkError::Netlink(ref e) if -e.raw_code() == libc::EACCES => match addr {
                     ipnet::IpNet::V6(_) => {
                         return Err(NetavarkError::wrap(
                             "failed to add ipv6 address, is ipv6 enabled in the kernel?",
@@ -449,9 +449,9 @@ impl Socket {
                 }
 
                 match rx_packet.payload {
-                    NetlinkPayload::Done => return Ok(result),
-                    NetlinkPayload::Error(e) | NetlinkPayload::Ack(e) => {
-                        if e.code != 0 {
+                    NetlinkPayload::Done(_) => return Ok(result),
+                    NetlinkPayload::Error(e) => {
+                        if e.code.is_some() {
                             return Err(e.into());
                         }
                         return Ok(result);
Index: netavark/src/network/vlan.rs
===================================================================
--- netavark.orig/src/network/vlan.rs
+++ netavark/src/network/vlan.rs
@@ -276,7 +276,7 @@ fn setup(
             Ok(_) => break,
 
             Err(err) => match err {
-                NetavarkError::Netlink(ref e) if -e.code == libc::EEXIST => {
+                NetavarkError::Netlink(ref e) if -e.raw_code() == libc::EEXIST => {
                     let random = Alphanumeric.sample_string(&mut rand::thread_rng(), 10);
                     let tmp_name = "mv-".to_string() + &random;
                     let mut opts = opts.clone();
